diff --git a/project/java/Globals.java b/project/java/Globals.java
index 2f2e8a77d..47eec2430 100644
--- a/project/java/Globals.java
+++ b/project/java/Globals.java
@@ -137,7 +137,7 @@ class Globals
 	public static int MultitouchGestureSensitivity = 1;
 	public static int TouchscreenCalibration[] = new int[4];
 	public static String DataDir = new String("");
-	public static boolean VideoLinearFilter = true;
+	public static boolean VideoLinearFilter = false;
 	public static boolean MultiThreadedVideo = false;
 
 	public static boolean OuyaEmulation = false; // For debugging
