diff --git a/project/java/Settings.java b/project/java/Settings.java
index 9e6150a..09ade66 100644
--- a/project/java/Settings.java
+++ b/project/java/Settings.java
@@ -736,7 +736,7 @@ public class Settings
 			if( Globals.TouchscreenKeyboardTheme == 2 )
 				nativeSetupScreenKeyboardButtons(loadRaw(p, R.raw.sun));
 			if( Globals.TouchscreenKeyboardTheme == 3 )
-				nativeSetupScreenKeyboardButtons(loadRaw(p, R.raw.keen));
+				nativeSetupScreenKeyboardButtons(loadRaw(p, R.raw.liero));
 			if( Globals.TouchscreenKeyboardTheme == 4 )
 				nativeSetupScreenKeyboardButtons(loadRaw(p, R.raw.retro));
 			if( Globals.TouchscreenKeyboardTheme == 5 )
